modules/aleph/includes/aleph.availability.inc

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * @file
+ * Handles availability information from the library system.
+ */
+
+use Drupal\aleph\Aleph\AlephMaterial;
+use Drupal\aleph\Aleph\Handler\AlephMaterialHandler;
+
+/**
+ * Implements provider availability, items.
+ *
+ * @param array $provider_ids
+ *   Array of ting object ids (faust).
+ *
+ * @return array
+ *   Availability info.
+ */
+function aleph_availability_items(array $provider_ids) {
+  $items = array();
+  $material_handler = new AlephMaterialHandler(aleph_client());
+
+  foreach ($provider_ids as $provider_id) {
+    $material = new AlephMaterial();
+    $material->setId($provider_id);
+    $items[] = $material_handler->getHoldings($material);
+  }
+
+  $result = array();
+  foreach ($items as $item) {
+    $result[$item->getId()] = array(
+      'available' => $item->isAvailable(),
+      'reservable' => $item->isReservable(),
+    );
+  }
+
+  return $result;
+}
+
+/**
+ * Implements provider availability, holdings.
+ *
+ * @param array $provider_ids
+ *   Array of ting object ids (faust).
+ *
+ * @return array
+ *   Holdings info.
+ */
+function aleph_availability_holdings(array $provider_ids) {
+  $data = array();
+  $results = array();
+  $sub_libraries = array();
+
+  try {
+    $material_handler = new AlephMaterialHandler(aleph_client());
+  }
+  catch (Exception $e) {
+    watchdog_exception('aleph', $e);
+    return [];
+  }
+
+  foreach ($provider_ids as $provider_id) {
+    $material = new AlephMaterial();
+    $material->setId($provider_id);
+    $items = $material_handler->getHoldings($material);
+
+    $reserved_count = 0;
+    $available_count = 0;
+
+    foreach ($items as $item) {
+      // Count the total amount on sub-libraries and see if it's available or
+      // reservable.
+      if (array_key_exists($item->getSubLibraryCode(), $sub_libraries)) {
+        $sub_libraries[$item->getSubLibraryCode()]['total_count']++;
+        $sub_libraries[$item->getSubLibraryCode()]['available'] = $item->isAvailable();
+        $sub_libraries[$item->getSubLibraryCode()]['reservable'] = $item->isReservable();
+      }
+      else {
+        $sub_libraries[$item->getSubLibraryCode()] = array(
+          'placement' => array($item->getSubLibrary(), $item->getCollection()),
+          // Reference count not available in Aleph.
```

Comment:
Not sure, but there's nothing like it available in Aleph and they didn't seem to know what it was and not all providers implements it.